src/java/org/apache/cassandra/io/storage/StorageProvider.java

```diff
@@ -254,5 +270,13 @@ public FileHandle.Builder fileHandleBuilderFor(IndexDescriptor descriptor, Index
             }
             return new FileHandle.Builder(file).mmapped(true);
         }
+
+        @Override
+        public FileHandle.Builder flushTimeFileHandleBuilderFor(IndexDescriptor descriptor, IndexComponent component, IndexContext context)
+        {
+            // By default, no difference between accesses "at flush time" and "at query time", but subclasses may need
```

Comment:
How would this be changing on the CNDB side?

Reply:
My idea (mainly because that's the simplest option I can think of) was that in cndb, this new method would use mmap, but the other one (`fileHandleBuilderFor`, which is use at query time) will use non mmap (which effectively mean it willl end up in the custom `FileChannel.read` call that requests NSS). There is probably other ways to do the same, but the idea is basically that by having a different methods, we can make sure the "flush time" version reads from local disk (since nothing has been uploaded yet), but the "query time" version reads from NSS.